core/azure-core/src/main/java/com/azure/core/implementation/configuration/Configuration.java

```diff
@@ -0,0 +1,149 @@
+     */
+    public String get(String name) {
+        return configurations.get(name);
+    }
+
+    /**
+     * Gets the value of the configuration converted to {@code T}.
+     *
+     * If no configuration is found the default is returned.
+     *
+     * @param name Name of the configuration.
+     * @param defaultValue Value to return if the configuration isn't found.
+     * @param <T> Generic type that the configuration is converted to if found.
+     * @return The converted configuration if found, otherwise the default value is returned.
+     */
+    public <T> T get(String name, T defaultValue) {
+        return convertOrDefault(get(name), defaultValue);
+    }
+
+    /**
+     * Gets the converted value of the configuration.
+     *
+     * @param name Name of the configuration.
+     * @param converter Converter used to map the configuration to {@code T}.
+     * @param <T> Generic type that the configuration is converted to if found.
+     * @return The converted configuration if found, otherwise null.
+     */
+    public <T> T get(String name, Function<String, T> converter) {
+        String value = get(name);
+        if (ImplUtils.isNullOrEmpty(value)) {
+            return null;
+        }
+
+        return converter.apply(get(name));
+    }
+
+    /**
+     * Adds the configuration.
+     *
+     * @param name Name of the configuration.
+     * @param value Value of the configuration.
+     */
+    public void put(String name, String value) {
+        configurations.put(name, value);
+    }
+
+    /**
+     * Removes the configuration.
+     *
+     * @param name Name of the configuration.
+     * @return If the configuration was removed the value of it, otherwise null.
+     */
+    public String remove(String name) {
+        return configurations.remove(name);
+    }
+
+    /**
+     * Determines if the configuration exists.
+     *
+     * @param name Name of the configuration.
+     * @return True if the configuration exists, otherwise false.
+     */
+    public boolean contains(String name) {
+        return configurations.containsKey(name);
+    }
+
+    /**
+     * @return a clone of the Configuration object.
+     */
+    @SuppressWarnings("CloneDoesntCallSuperClone")
+    public Configuration clone() {
+        return new Configuration(configurations);
+    }
+
+    /**
+     * Attempts to convert the configuration value to {@code T}.
+     *
+     * If the value is null or empty then the default value is returned.
+     *
+     * @param value Configuration value retrieved from the map.
+     * @param defaultValue Default value to return if the configuration value is null or empty.
+     * @param <T> Generic type that the value is converted to if not null or empty.
+     * @return The converted configuration, if null or empty the default value.
+     */
+    @SuppressWarnings("unchecked")
+    static <T> T convertOrDefault(String value, T defaultValue) {
```

Comment:
I don't think this needs to be a static method?

Reply:
Made the method part of the instance